### PR TITLE
sat extension

### PR DESCRIPTION
**Pull request by matthewhanson**
**Related Issue(s):** #

#216 
#356 
#555 

**Proposed Changes:**

Consolidates satellite related fields from sar and eo extensions into a single sat extension

1. New sat extension proposal
2. remove fields from eo and sar that are in new proposal

**PR Checklist:**

- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: extensions/sat/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+| orbit_state        | string  | The state of the orbit. Either `ascending` or `descending` for polar orbiting satellites, or `geostationary` for geosynchronous satellites |
+| relative_orbit        | integer       | The relative orbit number at the time of acquisition. |
+
+**platform** is the unique name of the specific platform the instrument is attached to. For satellites this would 
+be the name of the satellite, whereas for drones this would be a unique name for the drone. Examples include 
+`landsat-8` (Landsat-8), `sentinel-2a` and `sentinel-2b` (Sentinel-2), `terra` and `aqua` (part of NASA EOS, 
+carrying the MODIS instruments), `mycorp-uav-034` (hypothetical drone name), and `worldview02` 
+(Maxar/DigitalGlobe WorldView-2).
+
+**instruments** is an array of all the sensors used in the creation of the data. For example, data from the Landsat-8 platform is collected with the OLI sensor as well as the TIRS sensor, but the data is distributed together so would be specified as ['oli', 'TItirsRS']. Other instrument examples include `msi` (Sentinel-2), `aster` (Terra), and `modis` (Terra and Aqua).
```
Comment (m-mohr) [inline extensions/sat/README.md]: TItirsRS?

**Comment by m-mohr** [comment]
I have concerns about several of the changes here.

First of all, the SAR extension was written in a consensus process with I think about 6 to 8 parties involved. Changing some of the decisions should also go through a broader review process again. For example, pass_direction seems to be a commonly used term in SAR, but orbit state is something new I have rarely heard before. absolute_orbit was requested from SAR folks (see related issues/PRs) and what is the reason to remove it again?

I'm also not sure whether we are moving too many fields to sat. The sun angles are not so relevant for SAR and could remain in EO, I think.

Also, we were trying to remove the replication of instrument(s), platform etc. But now it would be in sat and aerial again so it is duplicated again.

Don't get me wrong: I think aligning SAR and EO makes sense and I appreciate the effort going into this, but I'm afraid of changing too many things so close to 1.0. This is bringing back many things that were above proposal state back into the proposal phase just before 0.9 and a close 1.0. So we should be careful here as this will also break many implementations and catalogs.

**Comment by matthewhanson** [comment]
orbit state is the same as pass direction, and it is used in both EO and SAR.  Orbit state is just a more general term as it works for geostationary orbits as well, not just polar. I'm fine with keeping it as 'pass_direction', but it should still have the 3 states: ascending, descending, geostationary

I've been using the SAR extension quite a bit and I think that a lot of the fields that were initially included are not fields important for search at all, and seem to be more "let's put all SAR related fields in here".  I reached out to SAR folks and no one could think of any reason at all for needing absolute orbit.

The `sat` extension is intended to conceptually include fields related to positioning and geometry.  It would seem a shame to keep some of those fields in EO when this could be a pretty tightly defined extension.

I would recommend also dropping the prefix in Aerial, just hadn't gotten to commenting on that PR yet.

I get your concerns overall, but my main worry for a while now is that we would be stuck with bloated extensions that seemed to be more of a catch-all then a clearly defined extension, like EO currently. I've been wanting to make these changes but they've always been pushed off to the next version.   I'm less concerned overall with breaking existing catalogs, as we are still pre 1.0 and I think that's always been a risk. I think it's more important to get this right than worry about backwards compatibility.

**Comment by m-mohr** [comment]
> orbit state is the same as pass direction, and it is used in both EO and SAR. Orbit state is just a more general term as it works for geostationary orbits as well, not just polar. I'm fine with keeping it as 'pass_direction', but it should still have the 3 states: ascending, descending, geostationary

I'm fine with both if all communities are used to these terms. We should ask members of the communities about that, I guess?

> I've been using the SAR extension quite a bit and I think that a lot of the fields that were initially included are not fields important for search at all, and seem to be more "let's put all SAR related fields in here". I reached out to SAR folks and no one could think of any reason at all for needing absolute orbit.

absolute_orbit is arguably one of the fields that could probably be removed as it's really not really well suited for search. What else do you think are non-searchable fields?

> The `sat` extension is intended to conceptually include fields related to positioning and geometry. It would seem a shame to keep some of those fields in EO when this could be a pretty tightly defined extension.

Fair enough. This should be explained in the extension description.

On the other hand, it feels strange that there are multiple instrument related fields in SAR, but that actual instrument is not defined there and SAR doesn't require sat to be used.

> I would recommend also dropping the prefix in Aerial, just hadn't gotten to commenting on that PR yet.

That would lead to having multiple potential definitions per field name, which was (usually) not the case before. So to know which field exactly it is, I would have to check stac_extensions. That sounds too complex.

> I get your concerns overall, but my main worry for a while now is that we would be stuck with bloated extensions that seemed to be more of a catch-all then a clearly defined extension, like EO currently.

Fair point, but IMHO has nothing to do with moving fields to a sat extension. Instead we could just clean the extensions.

> I think it's more important to get this right than worry about backwards compatibility.

The question is whether we get this right with this approach. I'm not yet convinced at all, especially as we duplicate fields again in aerial and sat and some fields in sat don't apply to all use cases. At the moment it feels like the previous approach was cleaner as there was just one extension per use case which you had to look at. On the other hand we better separate sat from aerial...

**Comment by m-mohr** [comment]
This should now have a sat prefix, right?

**Comment by matthewhanson** [comment]
Should it?  
Are we doing core extensions - or will all extensions move out of this repo except the ones that move into the Item spec?

I would say that if prefixes are not required, then I wouldn't use one here, the field names are descriptive enough on their own.

File: extensions/sat/json-schema/schema.json [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  "definitions": {
+    "eo": {
+      "type": "object",
+      "required": [
+        "properties"
+      ],
+      "properties": {
+        "properties": {
+          "type": "object",
+          "required": [
```
Comment (matthewhanson) [inline extensions/sat/json-schema/schema.json]: @m-mohr is this sufficient for requiring the fields from the core Item?

**Comment by matthewhanson** [comment]
Updated to use a `sar` prefix, for now. We can discuss if it makes sense to remove the prefix for this extension

File: extensions/sat/json-schema/schema.json (hunk 2 above)
Comment (m-mohr) [reply extensions/sat/json-schema/schema.json]: I think that should work, let'y try it out tomorrow in a validator.
Are these requirements specified in the written spec?

File: extensions/sar/examples/envisat.json [hunk 3]
```diff
@@ -21,8 +23,12 @@
...
   "properties": {
     "datetime": "2009-05-20T02:40:46.763325Z",
-    "dtr:start_datetime": "2009-05-20T02:40:01.042784Z",
-    "dtr:end_datetime": "2009-05-20T02:41:31.483865Z",
+    "start_datetime": "2009-05-20T02:40:01.042784Z",
+    "end_datetime": "2009-05-20T02:41:31.483865Z",
+    "platform": "envisat",
+    "constellation": "envisat",
+    "instruments": ["asar"],
+    "pass_direction": "descending",
```
Comment (m-mohr) [inline extensions/sar/examples/envisat.json]: Prefix missing.

File: extensions/sar/examples/sentinel1.json [hunk 4]
```diff
@@ -28,6 +29,8 @@
     "platform": "sentinel-1a",
     "constellation": "sentinel-1",
     "instruments": ["c-sar"],
+    "pass_direction": "ascending",
+    "relative_orbit": 33,
```
Comment (m-mohr) [inline extensions/sar/examples/sentinel1.json]: Prefix missing

File: extensions/sat/json-schema/schema.json [hunk 5]
```diff
@@ -0,0 +1,76 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "schema.json#",
+  "title": "Sat Extension",
+  "description": "STAC Sat Extension to a STAC Item.",
+  "allOf": [
+    {
+      "$ref": "../../../item-spec/json-schema/item.json#/definitions/core"
```
Comment (m-mohr) [inline extensions/sat/json-schema/schema.json]: You can remove `#/definitions/core`

File: extensions/sat/json-schema/schema.json [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    },
+    {
+      "$ref": "#/definitions/sat"
+    },
+    {
+      "$ref": "../instrument/json-schema/schema.json"
+    }
+  ],
+  "definitions": {
+    "eo": {
```
Comment (m-mohr) [inline extensions/sat/json-schema/schema.json]: must be sat

**Pull request merged**